File: 0747-largest-number-at-least-twice-of-others/0747-largest-number-at-least-twice-of-others.cpp
class Solution {
public:
    int dominantIndex(vector<int>& nums) {
        int max_val = -1;
        int secondMax_val = -1;
        int maxIdx = -1;

        for(int i = 0; i < nums.size(); i++){
            if(nums[i] > max_val){
                secondMax_val = max_val;
                max_val = nums[i];
                maxIdx = i;
            }else if(nums[i] > secondMax_val){
                secondMax_val = nums[i];
            }
        }
        return (max_val >= 2 * secondMax_val) ? maxIdx : -1;
    }
};

File: 0747-largest-number-at-least-twice-of-others/0747-largest-number-at-least-twice-of-others.py
class Solution:
    def dominantIndex(self, nums: lis[int]) -> int:
        # m = max(nums)
        # if all(m >= 2 * x for x in nums if x != m):
        #     return nums.index(m)
        # return -1

        max_val = max(nums)
        max_idx = nums.index(max_val)

        for i , num in enumerate(nums):
            if i != max_idx and max_val < 2 * num:
                return -1
        return max_idx